### PR TITLE
GH-127: add mime type check for image upload

### PR DESCRIPTION
**Pull request by sebam2k4**
Adds a simple mime type check and restricts image upload to JPG, JPEG, and PNG (file extension check really).

Would this be sufficient as a fix? Alternatively, could look into another solution: determining the image type by inspecting and comparing the first few bytes to known jpg/jpeg/png file signatures (Magic Bytes).

### CODE REVIEW

**Comment by clawfire** [comment]
Hey thanks for the PR. 
I think we can also add some control on the `<input />` tag so most of the people will not be able to even select an unsupported file type 
```html
<input type="file" id="qrfile" style="display: none;" accept="image/*">
``` 
should be changed to something like
```html
<input type="file" id="qrfile" style="display: none;" accept="image/jpg,image/jpeg,image/png">
```

If you can add this to your PR, we'll have both browser and JS check 😃

File: src/index.js [hunk 1]
```diff
@@ -294,20 +294,25 @@ window.addEventListener('load', function() {
 
   $('#scanAnother').on('click', () => {
     navigateTo('scan');
-  })
+  });
```
Comment (clawfire) [inline src/index.js:297]: in our case it's isn't really necessary but I don't mind 😁 
Some contributors use it, some do not. I personally prefer with 😁

**Review by clawfire — APPROVED** [review]
Just check the comment I made on the Conversation, I think it's better to have the browser also performing the check but otherwise thanks a LOT for your contribution 👍🏼

**Comment by sebam2k4** [comment]
Nice one, I've specified the accepted mime types for the input. This is really nice as it actually creates an initial filter for the image selection.

Was reading a little bit more about mime types and turns out image/JPG is actually invalid so I removed it.

image/JPEG actually includes all of the following extensions: .jpg, .jpeg, .jfif, .pjpeg, .pjp

**Review by clawfire — APPROVED** [review]

**Pull request merged**